_posts/2018-09-06-stateless-microservices.md

```diff
@@ -0,0 +1,272 @@
+
+Functions are emphemeral because at any point they should be able to be replaced by any similar variant. This property means that all functions can be managed in the same way - from their health-check to their lifecycle, logging and monitoring.
+
+Functions auto-scale either from a minimum replica count to a maximum count, or even down to zero and back again to match demand or to conserve resources.
+
+Functions are single purposed, but common sense must prevail. We do not need to create a new FaaS function every time we write `function x() {}` or `def x:` in our code. A single purpose may be "CRUD for an employee record", "format this IBAN" or "identify the contents of this image using a machine learning model".
+
+If you've written at least one serverless function then you'll be aware that the entrypoint is often abstracted away from you. You just write a handler and you're done - any dependencies are then expressed in a common format such as a `package.json` for Node.js or a `requirements.txt` file for Python.
+
+Here's an example of a Node.js function:
+
+```
+"use strict"
+
+module.exports = (context, callback) => {
+    callback(undefined, {status: "done"});
+}
+```
+
+As a developer you should have no knowledge of the mechanics used to bootstrap the handler - this is the boring, repetitive detail that you can off-load to a FaaS framework or toolkit.
+
+We see a similar pattern of abstraction and off-loading of boring details with microservices frameworks like [Sinatra](http://sinatrarb.com), [Django](https://www.djangoproject.com) and [Express.js](https://expressjs.com). One size rarely fits all needs, so several options exist for each language or runtime.
+
+## What's a stateless microservice?
+
+A stateless microservice is a microservice which can be deployed as if it were a FaaS Function and managed by a FaaS framework or Platform such as OpenFaaS. For this reason no special routes, flags or filters are needed in the OpenFaaS CLI, Gateway API or UI. 
+
+As far as the OpenFaaS components are concerned a function is written in any language and packaged in a Docker image. It must serve content over HTTP on port 8080 and write a lock file to `/tmp/.lock`. If your function or service detects it is unhealthy then you can remove the lock file and OpenFaaS will restart/reschedule your function.
+
+OpenFaaS has a Ruby language template which can be used to create Ruby FaaS functions. A Ruby stateless microservice is a Ruby microservice created with [Ruby on Rails](https://rubyonrails.org), Sinatra or some other Ruby microservice framework. The primary difference is that you have more work to do than with a FaaS function. Now you have to manage your own Dockerfile, health-checks and routes.
+
+Sinatra is a DSL or framework for Ruby for building microservices rapidly.
```

Comment:
... for Ruby that enables micro services to be rapidly built

Trying to avoid the double “for”

Reply:
Good 1